src/dsu.rs

```diff
@@ -1 +1,88 @@
+// Implement (union by size) + (path compression)
+// Reference:
+// Zvi Galil and Giuseppe F. Italiano,
+// Data structures and algorithms for disjoint set union problems
+pub struct Dsu {
+    n: usize,
+    // root node: -1 * component size
+    // otherwise: parent
+    parent_or_size: Vec<i32>,
+}
 
+impl Dsu {
+    pub fn new(size: usize) -> Self {
```

Comment:
It is worth mentioning that `1 <= size < 2^31`, which is also missing in the original ACL (that takes `size` as `int`).
Could you mention it in the comment?

Reply:
According to the ACL documentation, it's 0<=n<=10^8, so is it okay to write like that?

Reply:
I overlooked the documentation. 🙇‍♂️ 
I think it's fine.

Reply:
I added constraints.